### PR TITLE
2211 custom callout view

### PR DESCRIPTION
**Pull request by RomainQuidet**
Adding a callback to ask user for a custom callout view following mapbox protocol.
Demo available.

### CODE REVIEW

**Comment by RomainQuidet** [comment]
rebased to resolve conflicts

**Comment by RomainQuidet** [comment]
rebased again

File: include/mbgl/ios/MGLCalloutViewProtocol.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#import <Foundation/Foundation.h>
+
+@protocol MGLCalloutViewDelegate;
+
+@protocol MGLCalloutViewProtocol <NSObject>
+
+@property (nonatomic, copy) NSString *title, *subtitle;
```
Comment (1ec5) [inline include/mbgl/ios/MGLCalloutViewProtocol.h]: Please audit this header for nullability and use the `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` macros to tell the compiler you’ve done so. For Xcode 6.4 compatibility, you’ll need to `#import "MGLTypes.h"`.

File: ios/app/MBXViewController.mm [hunk 2]
```diff
@@ -261,6 +264,10 @@ - (void)actionSheet:(UIActionSheet *)actionSheet didDismissWithButtonIndex:(NSIn
     }
     else if (buttonIndex == actionSheet.firstOtherButtonIndex + 9)
     {
+        [self presentAnnotationWithCustomCallout];
+    }
+    else if (buttonIndex == actionSheet.firstOtherButtonIndex + 10)
```
Comment (1ec5) [inline ios/app/MBXViewController.mm]: The next condition needs to become `+ 11` and so on.

**Comment by 1ec5** [comment]
Thanks for contributing this feature and for your patience. Overall, this PR looks good. Feel free to also update CHANGELOG.md with a reference to this PR.

File: include/mbgl/ios/MGLCalloutViewProtocol.h (hunk 1 above)
Comment (RomainQuidet) [inline include/mbgl/ios/MGLCalloutViewProtocol.h]: You're right, done!

File: ios/app/MBXViewController.mm (hunk 2 above)
Comment (RomainQuidet) [inline ios/app/MBXViewController.mm]: This needs a refactor as I've done on another pull request to simplify the indexes!. Anyway, done.

**Comment by RomainQuidet** [comment]
@1ec5 branch rebased for last master version with all your review points. Thanks !

**Comment by 1ec5** [comment]
The saga continues in #3456.
